core/src/main/java/org/apache/iceberg/DataTableScan.java

```diff
@@ -87,7 +87,7 @@ public CloseableIterable<FileScanTask> doPlanFiles() {
       manifestGroup = manifestGroup.ignoreResiduals();
     }
 
-    if (dataManifests.size() > 1 && shouldPlanWithExecutor()) {
+    if (shouldPlanWithExecutor() && (dataManifests.size() > 1 || deleteManifests.size() > 1)) {
```

Comment:
Data and deletes are planned separately. If we are scanning 1 data manifest, it does not mean we shouldn't parallelize the planing of deletes.

```
ManifestGroup planWith(ExecutorService newExecutorService) {
  this.executorService = newExecutorService;
  deleteIndexBuilder.planWith(newExecutorService);
  return this;
}
```